includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -285,6 +287,56 @@ public function process_redirect_payment( $order_id, $intent_id, $save_payment_m
 		}
 	}
 
+	/**
+	 * Set formatted readable payment method title for order,
+	 * using payment method details from accompanying charge.
+	 *
+	 * @param WC_Order $order WC Order being processed.
+	 * @param array    $payment_method_details Array of payment method details from charge.
+	 */
+	public function set_payment_method_title_for_order( $order, $payment_method_details ) {
+		if ( empty( $payment_method_details ) ) {
+			return;
+		}
+
+		switch ( $payment_method_details['type'] ) {
+			case 'card':
+				$details       = $payment_method_details['card'];
+				$funding_types = [
+					'credit'  => __( 'credit', 'woocommerce-payments' ),
+					'debit'   => __( 'debit', 'woocommerce-payments' ),
+					'prepaid' => __( 'prepaid', 'woocommerce-payments' ),
+					'unknown' => __( 'unknown', 'woocommerce-payments' ),
+				];
+
+				$payment_method_title = sprintf(
+					// Translators: %1$s card brand, %2$s card funding (prepaid, credit, etc.).
+					__( '%1$s %2$s card', 'woocommerce-payments' ),
+					ucfirst( $details['network'] ),
+					$funding_types[ $details['funding'] ]
+				);
+				break;
```

Comment:
This mirrors the formatting from the transaction details page [over here](https://github.com/Automattic/woocommerce-payments/blob/develop/client/payment-details/payment-method/card/index.js#L51).